crates/core/c8y_api/src/json_c8y.rs

```diff
@@ -143,7 +143,7 @@ impl TryFrom<ThinEdgeEvent> for C8yCreateEvent {
     type Error = SMCumulocityMapperError;
 
     fn try_from(event: ThinEdgeEvent) -> Result<Self, SMCumulocityMapperError> {
-        let event_type = event.name;
+        let mut event_type = event.name;
```

Comment:
It's not so easy to see what exactly this method is doing, as there is a mix of mutable states, matches and if statements.

Reply:
I can re-write :+1:

Reply:
Refactored.